web-src/src/components/ApplicationProvider.js

```diff
@@ -18,48 +18,99 @@ import { ExpressSdkService } from '../services/ExpressSdkService.js';
 } from '../state/PromptTemplatesState.js';
 import { TargetService } from '../services/TargetService.js';
 import { CsvParserService } from '../services/CsvParserService.js';
 import { FeatureFlagsService } from '../services/FeatureFlagsService.js';
+import { AemService } from '../services/AemService.js';
+import { contentFragmentState } from '../state/ContentFragmentState.js';
 
-const APP_VERSION = process.env.REACT_APP_VERSION || 'unknown';
+export const APP_VERSION = process.env.REACT_APP_VERSION || 'unknown';
+
+export const RUN_MODE_DEFAULT = 'default';
+export const RUN_MODE_CF = 'content-fragments';
 
 const FEATURE_FLAGS_PROJECT_ID = 'aem-generate-variations';
 
 const COMPLETE_ACTION = 'complete';
 const FEEDBACK_ACTION = 'feedback';
 const TARGET_ACTION = 'target';
 const CSV_PARSER_ACTION = 'csv';
+const CF_ACTION = 'cf';
 
 export const ApplicationContext = React.createContext(undefined);
 
+export function parseUrlParameters() {
+  const searchParams = new URLSearchParams(window.location.search);
+  return {
+    aemHost: `https://${searchParams.get('aemHost')}`,
+    fragmentId: searchParams.get('fragmentId'),
+  };
+}
+
+function createAemService(aemHost, accessToken) {
+  return new AemService({
+    aemHost,
+    cfEndpoint: actions[CF_ACTION],
+    accessToken,
+  });
+}
+
 export const ApplicationProvider = ({ children }) => {
   const { user, done } = useShellContext();
+  const setBundledPromptTemplates = useSetRecoilState(bundledPromptTemplatesState);
   const setCustomPromptTemplates = useSetRecoilState(customPromptTemplatesState);
+  const setContentFragment = useSetRecoilState(contentFragmentState);
   const [applicationContext, setApplicationContext] = useState(undefined);
 
   useEffect(() => {
     if (!user) {
       return;
     }
 
-    console.debug('Reading custom prompt templates...');
-    readCustomPromptTemplates().then((templates) => {
-      setCustomPromptTemplates(templates);
-    });
-
     const createApplicationContext = async () => {
       console.debug('Creating application context...');
 
       // Create the feature flags service first, as it might be used by other services
       const featureFlagsService = await FeatureFlagsService.create(FEATURE_FLAGS_PROJECT_ID);
 
+      const { aemHost, fragmentId } = parseUrlParameters();
+      console.log(`AEM Host: ${aemHost}`);
+      console.log(`Fragment ID: ${fragmentId}`);
+
+      const runMode = fragmentId ? RUN_MODE_CF : RUN_MODE_DEFAULT;
+      console.log(`Run Mode: ${runMode}`);
+
+      console.debug('Reading bundled prompt templates...');
+      setBundledPromptTemplates(parseBundledPromptTemplates(runMode));
+
+      console.debug('Reading custom prompt templates...');
+      readCustomPromptTemplates(runMode, RUN_MODE_DEFAULT).then((templates) => {
+        setCustomPromptTemplates(templates);
+      });
+
+      const aemService = aemHost ? createAemService(aemHost, user.imsToken) : undefined;
+
+      if (aemService && fragmentId) {
```

Comment:
Can we put all AEM and CF stuff **into** the `AEMService`, and minimise changes to `ApplicationProvider`?
Can we set `setContentFragment` inside `AEMService`?

Sth like:

```
aemService: new AEMService({
  searchParams: new URLSearchParams(window.location.search),
  accessToken: user.imsToken,
}),
```

Considering this feature is not even behind the feature flag, we leak a lot of CF stuff into the global application scope.

Reply:
> Considering this feature is not even behind the feature flag

It is effectively behind a feature flag, as it is only enabled when specific parameters are present.